portlets/opencps-portlet/docroot/WEB-INF/src/org/opencps/accountmgt/service/persistence/CitizenFinderImpl.java

```diff
@@ -0,0 +1,209 @@
+						"lower(opencps_acc_citizen.fullName)", StringPool.LIKE,
+						true, keywords);
+			} else {
+				sql = StringUtil
+						.replace(
+								sql,
+								"AND (lower(opencps_acc_citizen.fullName) LIKE ? [$AND_OR_NULL_CHECK$])",
+								StringPool.BLANK);
+			}
+			if (accountStatus == -1) {
+				sql = StringUtil.replace(sql,
+						"AND (opencps_acc_citizen.accountStatus = ?)",
+						StringPool.BLANK);
+			}
+
+			sql = CustomSQLUtil.replaceAndOperator(sql, andOperator);
+
+			SQLQuery q = session.createSQLQuery(sql);
+			q.setCacheable(false);
+			q.addEntity("Citizen", CitizenImpl.class);
+
+			QueryPos qPos = QueryPos.getInstance(q);
+
+			qPos.add(groupId);
+
+			if (accountStatus != -1) {
+				qPos.add(accountStatus);
+			}
+
+			if (keywords != null && keywords.length > 0) {
+
+				qPos.add(keywords, 2);
+
+			}
+
+			return (List<Citizen>) QueryUtil.list(q, getDialect(), start, end);
+		} catch (Exception e) {
+			_log.error(e);
+		} finally {
+			session.close();
+		}
+		return null;
+	}
+
+	public int countCitizen(long groupId, String keywords, int accountStatus) {
+
+		String[] names = null;
+		boolean andOperator = false;
+
+		if (Validator.isNotNull(keywords)) {
+			names = CustomSQLUtil.keywords(keywords);
+		} else {
+			andOperator = true;
+		}
+
+		return _countCitizen(groupId, names, accountStatus, andOperator);
+	}
+
+	private int _countCitizen(long groupId, String[] keywords,
+			int accountStatus, boolean andOperator) {
+
+		/*
+		 * keywords = CustomSQLUtil .keywords(keywords, false);
+		 */
+
+		Session session = null;
+		try {
+			session = openSession();
+			String sql = CustomSQLUtil.get(COUNT_CITIZEN);
+
+			if (keywords != null && keywords.length > 0) {
+
+				sql = CustomSQLUtil.replaceKeywords(sql,
+						"lower(opencps_acc_citizen.fullName)", StringPool.LIKE,
+						true, keywords);
+			} else {
+				sql = StringUtil
+						.replace(
+								sql,
+								"AND (lower(opencps_acc_citizen.fullName) LIKE ? [$AND_OR_NULL_CHECK$])",
+								StringPool.BLANK);
+			}
+
+			if (accountStatus == -1) {
+				sql = StringUtil.replace(sql,
+						"AND (opencps_acc_citizen.accountStatus = ?)",
+						StringPool.BLANK);
+			}
+
+			sql = CustomSQLUtil.replaceAndOperator(sql, andOperator);
+			SQLQuery q = session.createSQLQuery(sql);
+			q.setCacheable(false);
+
+			q.addScalar(COUNT_COLUMN_NAME, Type.INTEGER);
+
+			QueryPos qPos = QueryPos.getInstance(q);
+
+			qPos.add(groupId);
+
+			if (accountStatus != -1) {
+				qPos.add(accountStatus);
+			}
+
+			if (keywords != null && keywords.length > 0) {
+
+				qPos.add(keywords, 2);
+
+			}
+			Iterator<Integer> itr = q.iterate();
+
+			if (itr.hasNext()) {
+				Integer count = itr.next();
+
+				if (count != null) {
+					return count.intValue();
+				}
+			}
+
+		} catch (Exception e) {
+			_log.error(e);
+		} finally {
+			session.close();
+		}
+
+		return 0;
+	}
+
+	private Log _log = LogFactoryUtil.getLog(CitizenFinderImpl.class);
```

Comment:
@huymq phần này sẽ không đặt try catch ở đây, tầng này để throw ra exception.

Nhiều phần viết ntn, sửa lại toàn bộ nhé.